syntax = "proto3";

package spaceship;

service Reporting {
    rpc GetSpaceship (Coordinates) returns (stream Spaceship);
}

message Coordinates {
    float x = 1;
    float y = 2;
    float z = 3;
    float t = 4;
    float e = 5;
}

enum Alignment {
    ALLY = 0;
    ENEMY = 1;
}

enum SpaceClass {
    CORVETTE = 0;
    FRIGATE = 1;
    CRUISER = 2;
    DESTROYER = 3;
    CARRIER = 4;
    DREADNOUGHT = 5;
}

message Officer {
    string first_name = 1;
    string last_name = 2;
    string rank = 3;
}

message Spaceship {
    Alignment alignment = 1;
    string name = 2;
    SpaceClass class = 3;
    float length = 4;
    int32 crew_size = 5;
    bool armed = 6;
    repeated Officer officers = 7;
}